ipn/conf.go

```diff
@@ -32,6 +32,8 @@ type ConfigVAlpha struct {
 	AdvertiseRoutes []netip.Prefix `json:",omitempty"`
 	DisableSNAT     opt.Bool       `json:",omitempty"`
 
+	AppConnector *AppConnectorPrefs `json:",omitempty"` // advertise app connector
```

Comment:
..if non-nil

Reply:
This does mean this AppConnectorPrefs type is now kinda locked in as stable. But Prefs (while not user visible and thus not necessarily stable) have historically stayed mostly stable.

Reply:
I notice that `AppConnectorPrefs` has a concrete `bool` field instead of `opt.Bool`. Currently it's kind of ok because we have this pointer as an indirection in between, but it will become inconsistent with other config bools in future if we add more fields to `AppConnectorPrefs`, because you won't be able to meaningfully differentiate between unset and false for that particular field (if you need to set other fields in `AppConnectorPrefs`).

Reply:
oh, I didn't even look yet. We should probably just make our own types here then.

Reply:
> but it will become inconsistent with other config bools in future if we add more fields to AppConnectorPrefs

This is for it to be consistent with the other field here that is a struct that has a bool field `AutoUpdatePrefs`. So set `AppConnectorPrefs`, unset `AppConnectorPrefs.Advertise` means that the field itself is set which it is (same as with `AutoUpdatePrefs.Check`